File: analytics/src/main/java/io/appmetrica/analytics/impl/db/connectors/SimpleDBConnector.java
package io.appmetrica.analytics.impl.db.connectors;

import android.database.sqlite.SQLiteDatabase;
import androidx.annotation.Nullable;
import io.appmetrica.analytics.impl.db.DatabaseStorage;
import io.appmetrica.analytics.logger.internal.DebugLogger;

public class SimpleDBConnector implements DBConnector {

    private static final String TAG = "[SimpleDBConnector]";

    private final DatabaseStorage mStorage;

    public SimpleDBConnector(DatabaseStorage storage) {
        mStorage = storage;
    }

    @Nullable
    public SQLiteDatabase openDb() {
        try {
            return mStorage.getWritableDatabase();
        } catch (Throwable ex) {
            DebugLogger.warning(TAG,"Something went wrong while opening database\n" + ex);
        }
        return null;
    }

    public void closeDb(@Nullable final SQLiteDatabase db) {}
}

File: analytics/src/main/java/io/appmetrica/analytics/impl/db/connectors/SimpleDBConnector.kt
package io.appmetrica.analytics.impl.db.connectors

import android.database.sqlite.SQLiteDatabase
import io.appmetrica.analytics.impl.db.DatabaseStorage
import io.appmetrica.analytics.logger.appmetrica.internal.DebugLogger

internal class SimpleDBConnector(private val storage: DatabaseStorage) : DBConnector {

    private val tag = "[SimpleDBConnector]"

    override fun openDb(): SQLiteDatabase? {
        return try {
            storage.writableDatabase
        } catch (ex: Throwable) {
            DebugLogger.warning(tag, "Something went wrong while opening database\n$ex")
            null
        }
    }

    override fun closeDb(db: SQLiteDatabase?) {
        // Do nothing
    }
}
